core/src/main/java/hudson/Functions.java

```diff
@@ -758,6 +758,20 @@ public static boolean isMultiline(String s) {
         return s.indexOf('\r') >= 0 || s.indexOf('\n') >= 0;
     }
 
+    /**
+     * Partially encodes a string into UTF-8 encoding. It uses {@link hudson.Util#encode} to escape
+     * non-ASCII characters in URL.
```

Comment:
Partial seems unnecessarily vague, and mentioning only UTF-8 seems misleading, given this percent-encodes the string (which are orthogonal).

How about:

```suggestion
     * Percent-encodes space and non-ASCII UTF-8 characters for use in URLs.
```